app/pathogen/arbovirus/analyze/recharts/change-in-median-seroprevalence-over-time-graph.tsx

```diff
@@ -0,0 +1,196 @@
+"use client";
+
+import { ArboContext } from "@/contexts/arbo-context";
+import React, { useContext } from "react";
+import { isAfter, isBefore, startOfYear, addYears, endOfYear, parseISO } from "date-fns";
+import {
+  typedGroupBy,
+  typedObjectEntries,
+  typedObjectFromEntries,
+} from "@/lib/utils";
+import { TimeInterval, doTimeIntervalsOverlap } from "@/lib/date-utils";
+import clsx from "clsx";
+import { SlantedTick, arbovirusesSF, convertArboSFtoArbo, median } from "../recharts";
+import { Bar, BarChart, CartesianGrid, ResponsiveContainer, Tooltip, XAxis, YAxis } from "recharts";
+import { pathogenColors } from "../../dashboard/(map)/ArbovirusMap";
+
+type GroupableIntoTimeBuckets = { groupingTimeInterval: TimeInterval };
+
+interface GroupDataPointsIntoTimeBucketsInput<
+  TDataPoint extends GroupableIntoTimeBuckets
+> {
+  dataPoints: TDataPoint[];
+  groupingBucketSize: {
+    years: number;
+  };
+}
+
+interface GroupDataPointsIntoTimeBucketsOutput<
+  TDataPoint extends GroupableIntoTimeBuckets
+> {
+  groupedDataPoints: {
+    interval: TimeInterval;
+    dataPoints: TDataPoint[];
+  }[];
+}
+
+interface GenerateBucketsForTimeIntervalInput {
+  earliestDate: Date;
+  latestDate: Date;
+  groupingBucketSize: {
+    years: number;
+  };
+}
+
+interface GenerateBucketsForTimeIntervalOutput {
+  buckets: TimeInterval[];
+}
+
+const generateBucketsForTimeInterval = (input: GenerateBucketsForTimeIntervalInput): GenerateBucketsForTimeIntervalOutput => {
+  let currentIntervalStartDate = startOfYear(input.earliestDate);
+  const buckets: TimeInterval[] = [];
+  
+  while(isBefore(currentIntervalStartDate, input.latestDate)) {
+    const currentIntervalEndDate = endOfYear(addYears(currentIntervalStartDate, input.groupingBucketSize.years));
+    const bucketToAdd = {
+      intervalStartDate: parseISO(currentIntervalStartDate.toISOString()),
+      intervalEndDate: parseISO(currentIntervalEndDate.toISOString())
```

Comment:
Scared of some JavaScript "copy-reference-instead-of-value"  thing so this is just my way of cloning the dates.